File: backend/MatrixMarket.Api/DTOs/OfferDtos.cs
namespace MatrixMarket.Api.DTOs;

public class CreateOfferRequest
{
    public int ListingId { get; set; }
    public decimal OfferedPrice { get; set; }
}

public class ScheduleTradeRequest
{
    public int OfferId { get; set; }
    public string CampusSpot { get; set; } = string.Empty;
    public DateTime MeetingTime { get; set; }
}

public class TradeMeetingResponse
{
    public int MeetingId { get; set; }
    public int OfferId { get; set; }
    public string CampusSpot { get; set; } = string.Empty;
    public DateTime MeetingTime { get; set; }
    public string ConfirmationCode { get; set; } = string.Empty;
    public bool IsConfirmed { get; set; }
    public string ListingTitle { get; set; } = string.Empty;
}

public class ConfirmTradeRequest
{
    public string ConfirmationCode { get; set; } = string.Empty;
}

// Returned by GET /api/v1/trades/mine so BOTH sides of a trade - not just the
// buyer who scheduled it - can see the meeting spot, time, and confirmation code.
public class MyTradeMeetingResponse
{
    public int MeetingId { get; set; }
    public int OfferId { get; set; }
    public string CampusSpot { get; set; } = string.Empty;
    public DateTime MeetingTime { get; set; }
    public string ConfirmationCode { get; set; } = string.Empty;
    public bool IsConfirmed { get; set; }
    public string ListingTitle { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public string Role { get; set; } = string.Empty; // "Buyer" or "Seller"
    public string OtherPartyName { get; set; } = string.Empty;
}

File: android/app/src/main/java/com/matrixmarket/app/data/remote/dto/OfferDtos.kt
package com.matrixmarket.app.data.remote.dto

data class CreateOfferRequest(
    val listingId: Int,
    val offeredPrice: Double
)

data class OfferResponse(
    val offerId: Int,
    val status: String
)

data class ScheduleTradeRequest(
    val offerId: Int,
    val campusSpot: String,
    val meetingTime: String // ISO-8601, e.g. 2026-09-20T14:00:00
)

data class TradeMeetingResponse(
    val meetingId: Int,
    val offerId: Int,
    val campusSpot: String,
    val meetingTime: String,
    val confirmationCode: String,
    val isConfirmed: Boolean,
    val listingTitle: String
)

data class ConfirmTradeRequest(
    val confirmationCode: String
)

// One row for the Meet tab's list - works for both sides of the trade, since
// "role" tells us whether this user is the Buyer or the Seller for this meeting.
data class MyTradeMeetingResponse(
    val meetingId: Int,
    val offerId: Int,
    val campusSpot: String,
    val meetingTime: String,
    val confirmationCode: String,
    val isConfirmed: Boolean,
    val listingTitle: String,
    val price: Double,
    val role: String, // "Buyer" or "Seller"
    val otherPartyName: String
)
